unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/Settings/Settings.cs

```diff
@@ -1,78 +1,64 @@
 using System;
 using DCL.Helpers;
 using DCL.Interface;
-using UnityEngine;
 using UnityEngine.Audio;
 
 namespace DCL.SettingsCommon
 {
-    public class Settings : Singleton<Settings>
+    public class Settings
     {
-        const string QUALITY_SETTINGS_KEY = "Settings.Quality";
-        const string GENERAL_SETTINGS_KEY = "Settings.General";
-        const string AUDIO_SETTINGS_KEY = "Settings.Audio";
+        public static Settings i { get; private set; }
+
         public event Action OnResetAllSettings;
+        
         public QualitySettingsData qualitySettingsPresets => qualitySettingsPreset;
-
-        public ISettingsRepository<QualitySettings> qualitySettings;
-        public ISettingsRepository<GeneralSettings> generalSettings;
-        public ISettingsRepository<AudioSettings> audioSettings;
-
-        private static QualitySettingsData qualitySettingsPreset = null;
-
-        public QualitySettingsData autoqualitySettings = null;
+        public QualitySettingsData autoQualitySettings;
         public QualitySettings lastValidAutoqualitySet;
 
-        private readonly BooleanVariable autosettingsEnabled = null;
+        public readonly ISettingsRepository<QualitySettings> qualitySettings;
+        public readonly ISettingsRepository<GeneralSettings> generalSettings;
+        public readonly ISettingsRepository<AudioSettings> audioSettings;
+        
+        private readonly QualitySettingsData qualitySettingsPreset;
+        private readonly BooleanVariable autoQualitySettingsEnabled;
         private readonly AudioMixer audioMixer;
 
-        public Settings()
+        private bool isDisposed;
+
+        public static void CreateSharedInstance(ISettingsFactory settingsFactory)
```

Comment:
I think we can transform this pattern into the service locator pattern when the new service locator PR is merged. 

With the new pattern, an interface would have to be extracted from this `Settings` class, and it should be like this:

On `unity-renderer`:
`Register<ISettings>( () => new Settings() )`

Then, on [SetupServices](https://github.com/decentraland/unity-renderer/blob/7e1e5971e1e97c98176245ce757dcd63e2d76636/unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/Scene/Main/Main.cs#L76) an alternative service locator can be built by using this line:

`serviceLocator.Register<ISettings>( () => new DesktopSettings() )`

(Here's the `ServiceLocatorFactory.CreateDefault()` [implementation](https://github.com/decentraland/unity-renderer/blob/14296c66f2e967d8339cad53317505ce0d4467bf/unity-renderer/Assets/Scripts/MainScripts/DCL/Environment/Factories/ServiceLocatorFactory/ServiceLocatorFactory.cs#L9))